src/views/Tenant/TenantView.js

```diff
@@ -0,0 +1,264 @@
+  const getTableData = () => [
+    {
+      key: "firstName",
+      label: "First Name",
+      value: tenant.firstName,
+      inputType: "text",
+    },
+    {
+      key: "lastName",
+      label: "Last Name",
+      value: tenant.lastName,
+      inputType: "text",
+    },
+    {
+      key: "phone",
+      label: "Phone",
+      value: tenant.phone,
+      inputType: "text",
+    },
+    {
+      key: "address",
+      label: "Property",
+      value: `${property.address}, ${property.city}, ${property.state}, ${property.zipcode}`,
+      inputType: "text",
+      comp: <div />,
+    },
+    {
+      key: "unit",
+      label: "Unit",
+      value: property.unit,
+      inputType: "text",
+      comp: <div />,
+    },
+  ];
+
+  /**
+   * When staff search input text changes, call API to find matching users.
+   */
+  useEffect(() => {
+    const loadStaff = async () => {
+      const staffResponse = await axios.post("/api/users/role", { userrole: 2, name: staffSearchText });
+      const foundStaff = await staffResponse.data;
+      const foundStaffChoices = getStaffChoice(foundStaff.users);
+      setStaffSearchResults(foundStaffChoices);
+    };
+    if (staffSearchText && staffSearchText.length > 0) {
+      loadStaff();
+    }
+  }, [staffSearchText]);
+
+  /**
+   * Handle staff search input
+   * @param {*} event
+   */
+  const handleChangeSearch = (event) => {
+    const { value } = event.target;
+    if (!value || value.length === 0) {
+      setStaffSearchResults([]);
+      setStaffSearchText("");
+    } else {
+      setStaffSearchText(value);
+    }
+  };
+
+  /**
+   * Handle change in staff selections of search panel
+   * @param {*} selectedChoices
+   */
+  const handleChangeStaffSelections = (selectedChoices) => {
+    setStaffSelections(selectedChoices);
+  };
+
+  return (
+    <div className="manager__container">
+      {tenant && (
+        <div>
+          <div className="title__container">
+            <h2>
+              {tenant.firstName}
+              {" "}
+              {tenant.lastName}
+            </h2>
+            <button
+              className={`rounded${isEditing ? "--is-editing" : ""}`}
+              onClick={handleEditToggle}
+              disabled={isEditing}
+            >
+              <i className="fas fa-pen icon" />
+            </button>
+          </div>
+          <div className="manager__contact">
+            <h1 className="secondary-title">CONTACT</h1>
+            <div className="contact-details">
+              <ToggleEditTable
+                tableData={getTableData()}
+                validationSchema={validationSchema}
+                isEditing={isEditing}
+                submitHandler={onFormikSubmit}
+                cancelHandler={onCancelClick}
+              />
+            </div>
+          </div>
+
+          <div className="manager__contact">
+            <h1 className="secondary-title">JOIN STAFF</h1>
+          </div>
+
+          <div>
+            <SearchPanel
```

Comment:
While I love server-side search tools, it might be easier for the user to see the whole list (or like the first 5 items) on click of the search component (without typing anything), since I don't think it will be likely to have hundreds or thousands of results for this search bar.

We should probably shorten the width of the Search component like the mockup.

Reply:
I fixed the width and made an initial search call to load all staff. It will scroll when there are more results than would fill 200px height. If we aren't too concerned about this API call being too big or slow, I like the idea of letting the API do the filtering so you don't have to reload when data changes. But, we could code a client side onChange handler too. Would you prefer client side?

Reply:
Serverside is good! I think the changes you made should be fine